scripts/skills/perks/perk_rf_sneak_attack.nut

```diff
@@ -17,6 +17,78 @@ this.perk_rf_sneak_attack <- ::inherit("scripts/skills/skill", {
 		this.m.IsHidden = false;
 	}
 
+	function getTooltip()
+	{
+		local tooltip = this.skill.getTooltip();
+
+		if (this.getContainer().getActor().isArmedWithRangedWeapon())
+		{
+			tooltip.extend([
+				{
+					id = 10,
+					type = "text",
+					icon = "ui/icons/special.png",
+					text = "Against targets you have not previously attacked or who have not previously attacked you, gain:"
+				},
+				{
+					id = 10,
+					type = "text",
+					icon = "ui/icons/regular_damage.png",
+					text = ::MSU.Text.colorGreen("15%") + " increased damage"
+				},
+				{
+					id = 10,
+					type = "text",
+					icon = "ui/icons/direct_damage.png",
+					text = ::MSU.Text.colorGreen("+10%") + " armor penetration"
+				}
+			]);
+		}
+		else
+		{
+			tooltip.extend([
```

Comment:
I would check the main weapon at this point and print a warning if its reach is more than 4 instead of the following tooltip

Reply:
The effect is hidden unless a valid weapon is equipped. Which is why I didn't do that check here. It is done in `isEnabled`.